hugr-core/src/types/type_row.rs

```diff
@@ -168,15 +168,15 @@ impl From<TypeRow> for TypeRowRV {
     }
 }
 
-impl<const RV: bool> From<&'static [TypeBase<RV>]> for TypeRowBase<RV> {
+impl<RV: MaybeRV> From<&'static [TypeBase<RV>]> for TypeRowBase<RV> {
     fn from(types: &'static [TypeBase<RV>]) -> Self {
         Self {
             types: types.into(),
         }
     }
 }
 
-impl<const RV1: bool> From<TypeBase<RV1>> for TypeRowRV {
+impl<RV1: MaybeRV + Into<RowVariable>> From<TypeBase<RV1>> for TypeRowRV {
```

Comment:
Yes, this means we have a slightly-weird-looking `MaybeRV: From<NoRV> + Into<RowVariable>` but that does exactly capture the slightly-weird constraints we need for the conversions

Reply:
For a moment I even thought this might allow removing `into_` altogether, but we'd need to parameterize `where TypeBase<T>: Into Type` or something weird, and the same (or similar-but-different) in a bunch of other places. So I'm not 100% certain we can't remove `into_` but I'm not that optimistic. Can try a bit more (maybe after we merge this into #1138)